test/validate_templates.py

```diff
@@ -0,0 +1,63 @@
+from jinja2 import BaseLoader, Environment, meta, TemplateError
+from templates import TemplateCollection
+from utils import get_dataset_builder
+
+#
+# This script validates all the templates in the repository with simple syntactic
+# checks:
+#
+# 0. Are all templates parsable YAML?
+# 1. Do all templates parse in Jinja and are all referenced variables in the dataset schema?
+# 2. Does the template contain a prompt/output separator "|||" ?
+# 3. Are all names and templates within a data (sub)set unique?
+#
+
+# Sets up Jinja environment
+env = Environment(loader=BaseLoader)
+
+# Loads templates and iterates over each data (sub)set
+template_collection = TemplateCollection()
+for (dataset_name, subset_name) in template_collection.keys:
```

Comment:
you can also separate a test call per dataset. Otherwise the for loop will exit at the first error you encounter rather than throwing raising for all data (sub) sets.

An example here: https://github.com/huggingface/datasets/blob/master/tests/test_dataset_cards.py#L74